stdlib/asyncio/coroutines.pyi

```diff
@@ -24,4 +23,6 @@ if sys.version_info >= (3, 8):
     def iscoroutine(obj: object) -> TypeGuard[Coroutine[Any, Any, Any]]: ...
 
 else:
-    def iscoroutine(obj: object) -> TypeGuard[types.GeneratorType[Any, Any, Any] | Coroutine[Any, Any, Any]]: ...
+    def iscoroutine(
+        obj: object,
+    ) -> Annotated[TypeGuard[Coroutine[Any, Any, Any]], "can actually be a generator-style coroutine"]: ...
```

Comment:
I'd prefer to have a comment instead of using Annotated here, in the off-chance this interacts poorly with someone else's use of Annotated.